R/sample_grid.R

```diff
@@ -0,0 +1,109 @@
+##' Take a grid search produced by \code{\link{scan_beta_date}} and 
+##' sample \code{n_sample_pairs} from the parameter grid uses based
+##' on their probaility. For each parameter pair chosen, run particle
+##' filter with \code{num_particles} and sample 1 trajectory
+##' 
+##' @title Sample Grid Scan
+##' 
+##' @param scan_results Output of \code{\link{scan_beta_date}}.
+##' 
+##' @param n_sample_pairs Number of parameter pairs to be sampled. Integer.
+##'   Default = 10.
+##'   
+##' @param n_particles Number of particles. Positive Integer. Default = 100
+##'   
+##' @return List. First element is an array of trajectories, in which
+##'  first dimension is time, second is parameters chosen when sampling from 
+##'  the grid and the third dimension is alist of inputs.
+##' 
+##' @export
+##' @import furrr
+##' @importFrom utils tail
+sample_grid_scan <- function(scan_results, 
+                             n_sample_pairs = 10, 
+                             n_particles = 100) {
```

Comment:
n_trajectories would be more appropriate? if not I think it is confusing with the number of particles you run your particle filter with (which decide the "accuracy" of it)

Reply:
Wil resolve by adding to documentation to show that n_sample_pairs controls the number of trajectories.